commit 790a35bd65c0449bd8d60dae5053f97ab75c3ab5
Author: Timothy G Abbott <tabbott@mega-man.mit.edu>
Date:   Sat Apr 19 22:26:11 2008 -0400

    Don't leave new conffiles as .dpkg-new versions when the conffile has been diverted.
    
    * src/configure.c: (deferred_configure) Compare the diverted versions of conffiles.

diff --git a/src/configure.c b/src/configure.c
index cde2c68..e67f57e 100644
--- a/src/configure.c
+++ b/src/configure.c
@@ -85,6 +85,7 @@ void deferred_configure(struct pkginfo *pkg) {
 	struct stat stab;
 	enum conffopt what;
 	static const char *EMPTY_HASH = "-";
+	const char *usename;
 
 	if (pkg->status == stat_notinstalled)
 		ohshit(_("no package named `%s' is installed, cannot configure"),pkg->name);
@@ -173,7 +174,8 @@ void deferred_configure(struct pkginfo *pkg) {
 		varbufinit(&cdr);
 		varbufinit(&cdr2);
 		for (conff= pkg->installed.conffiles; conff; conff= conff->next) {
-			r= conffderef(pkg, &cdr, conff->name);
+			usename = namenodetouse(findnamenode(conff->name,fnn_nocopy),pkg)->name;
+			r= conffderef(pkg, &cdr, usename);
 			if (r == -1) {
 				conff->hash= EMPTY_HASH;
 				continue;
@@ -209,7 +211,7 @@ void deferred_configure(struct pkginfo *pkg) {
 				what= cfo_identical;
 			} else if (!strcmp(currenthash,NONEXISTENTFLAG) && fc_conff_miss) {
 				fprintf(stderr, _("\nConfiguration file `%s', does not exist on system.\n"
-							"Installing new config file as you request.\n"), conff->name);
+							"Installing new config file as you request.\n"), usename);
 				what= cfo_newconff;
 				useredited= -1;
 				distedited= -1;
@@ -233,9 +235,9 @@ void deferred_configure(struct pkginfo *pkg) {
 
 			debug(dbg_conff,
 					"deferred_configure `%s' (= `%s') useredited=%d distedited=%d what=%o",
-					conff->name, cdr.buf, useredited, distedited, what);
+					usename, cdr.buf, useredited, distedited, what);
 
-			what=promptconfaction(conff->name, cdr.buf, cdr2.buf, useredited, distedited, what);
+			what=promptconfaction(usename, cdr.buf, cdr2.buf, useredited, distedited, what);
 
 			switch (what & ~(cfof_isnew|cfof_userrmd)) {
 				case cfo_keep | cfof_backup:
@@ -281,7 +283,7 @@ void deferred_configure(struct pkginfo *pkg) {
 								pkg->name, cdr.buf, cdr2.buf, strerror(errno));
 					/* fall through */
 				case cfo_install:
-					printf(_("Installing new version of config file %s ...\n"),conff->name);
+					printf(_("Installing new version of config file %s ...\n"),usename);
 				case cfo_newconff:
 					strcpy(cdr2rest,DPKGNEWEXT);
 					trig_file_activate_byname(conff->name, pkg);
